app/controllers/sites_controller.rb

```diff
@@ -33,10 +33,9 @@ class SitesController < ApplicationController
 
   def index
     search = new_search
-
     search.name_start_with params[:name] if params[:name].present?
     search.alerted_search params[:_alert] if params[:_alert] == "true"
-    search.my_site_search current_user.id unless current_user.can_view_other? params[:collection_id]
+    search.my_site_search current_user.id if !current_user.is_guest && !current_user.can_view_other?(params[:collection_id])
```

Comment:
Seem searching my site is contrast with condition, I can search only when I can't view the other otherwise I can't